integrations/oci/objectstorage/src/main/java/io/helidon/integrations/oci/objectstorage/OciObjectStorageHealthCheck.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.objectstorage;
+
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.helidon.common.reactive.Single;
+import io.helidon.config.Config;
+import io.helidon.health.common.BuiltInHealthCheck;
+import io.helidon.integrations.common.rest.ApiOptionalResponse;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Liveness;
+
+import static io.helidon.common.http.Http.Status.OK_200;
+
+/**
+ * Liveness check for an OCI's ObjectStorage bucket. Reads configuration to
+ * obtain bucket name as well as OCI properties from '~/.oci/config'.
+ */
+@Liveness
+@ApplicationScoped // this will be ignored if not within CDI
+@BuiltInHealthCheck
+public final class OciObjectStorageHealthCheck implements HealthCheck {
+    private static final Logger LOGGER = Logger.getLogger(OciObjectStorageHealthCheck.class.getName());
+
+    private static final int DEFAULT_TIMEOUT_SECONDS = 10;
+
+    private String bucket;
+    private String namespace;
+    private int timeout;
+    private OciObjectStorageRx ociObjectStorage;
+
+    private OciObjectStorageHealthCheck() {
+        this.timeout = DEFAULT_TIMEOUT_SECONDS;
+        Config ociConfig = Config.create().get("oci");
```

Comment:
I guess this is intended to be used by CDI?
If that is the case, you can just use `@Inject` on the constructor, and use `@ConfigProperty` to inject all configuration and get rid of the system properties.
Using `Config.create()` outside of user code is not recommended and may produce unexpected results.